public/app/features/preferences/api.ts

```diff
@@ -0,0 +1,67 @@
+import { BaseQueryFn, createApi } from '@reduxjs/toolkit/query/react';
+import { lastValueFrom } from 'rxjs';
+
+import { BackendSrvRequest, getBackendSrv } from '@grafana/runtime';
+import { Preferences } from '@grafana/schema';
+
+interface RequestOptions extends BackendSrvRequest {
+  manageError?: (err: unknown) => { error: unknown };
+  showErrorAlert?: boolean;
+  body?: BackendSrvRequest['data'];
+}
+
+function createBackendSrvBaseQuery({ baseURL }: { baseURL: string }): BaseQueryFn<RequestOptions> {
+  async function backendSrvBaseQuery(requestOptions: RequestOptions) {
+    try {
+      const { data: responseData, ...meta } = await lastValueFrom(
+        getBackendSrv().fetch({
+          ...requestOptions,
+          url: baseURL + requestOptions.url,
+          showErrorAlert: requestOptions.showErrorAlert,
+          data: requestOptions.body,
+        })
+      );
+      return { data: responseData, meta };
+    } catch (error) {
+      return requestOptions.manageError ? requestOptions.manageError(error) : { error };
+    }
+  }
+
+  return backendSrvBaseQuery;
+}
+
+export interface UpdateRequestArg {
+  body: Preferences;
+}
+
+const URL = '/user/preferences';
```

Comment:
It looks like these endpoints are fully described in our OpenAPI spec, so I think we can generate this?

Edit `generate-rtk-apis.ts` to add a new output for `getUserPreferences` / `patchUserPreferences` / `updateUserPreferences` (and the org preferences as well?)

If we can generate it, then we don't have to have an opinion on the style of the output.